report/src/main/kotlin/com/hartwig/actin/report/pdf/tables/molecular/TreatmentRankingGenerator.kt

```diff
@@ -13,13 +13,48 @@ class TreatmentRankingGenerator(private val treatmentEvidenceRanking: TreatmentE
 
     override fun forceKeepTogether() = false
 
+    /*
     override fun contents(): Table {
         val table = Tables.createRelativeWidthCols(1f, 1f, 1f)
         val header = listOf("Treatment", "Events", "Score").map(Cells::createHeader)
         val cells = treatmentEvidenceRanking.ranking.sortedBy { it.score }.reversed().flatMap {
-            listOf(it.treatment, it.events.joinToString("\n"), DecimalFormat.getIntegerInstance().format(it.score))
+            listOf(it.treatment, it.events.toString(), DecimalFormat.getIntegerInstance().format(it.score))
         }.map(Cells::createContent)
         (header + cells).forEach(table::addCell)
         return table
     }
+    */
+
+    override fun contents(): Table {
+        val table = Tables.createRelativeWidthCols(1f, 1f, 1f)
+        val header = listOf("Event / Variant", "Treatment", "Score").map(Cells::createHeader)
+
+        // Group treatments by associated events
+        val groupedRankings = treatmentEvidenceRanking.ranking.groupBy { it.events }
+
+        // Partition groups into variants and event groups
+        val (variantGroups, eventGroups) = groupedRankings.entries.partition { (event, _) ->
+            event != null && event.matches(Regex(".*\\b(del|amp|fusion|mut|[A-Za-z]+\\d+)\\b.*", RegexOption.IGNORE_CASE))
+        }
+
+        // Combine groups with variants first
+        val orderedGroups = variantGroups + eventGroups
```

Comment:
So the idea is we give some ordered priority to variants over characteristics? Was that a request? Feels a bit arbitrary, I thought we should order by the group with the highest top score (or maybe total score)?